lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,164 @@
+
+                   this.teamDescriptionField = new FieldViews.TextareaFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Team Description (Required) *"),
+                       valueAttribute: 'description',
+                       editable: 'always',
+                       showMessages: false,
+                       descriptionMessage: gettext("A short description of the team to help other learners understand the goals or direction of the team (maximum 300 characters).")
+                   });
+
+                   this.optionalDescriptionField = new FieldViews.ReadonlyFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Optional Characteristics"),
+                       valueAttribute: 'optional_description',
+                       helpMessage: gettext("Help other learners decide whether to join your team by specifying some characteristics for your team. Choose carefully, because fewer people might be interested in joining your team if it seems too restrictive. You cannot change these characteristics after you create the team.")
+                   });
+
+                   this.teamLanguageField = new FieldViews.DropdownFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Language"),
+                       valueAttribute: 'language',
+                       required: false,
+                       showMessages: false,
+                       titleIconName: 'fa-comment-o',
+                       options: this.languages,
+                       helpMessage: gettext("The language that team members primarily use to communicate with each other.")
+                   });
+
+                   this.teamCountryField = new FieldViews.DropdownFieldView({
+                       model: new TeamModel(),
+                       title: gettext('Country'),
+                       valueAttribute: 'country',
+                       required: false,
+                       showMessages: false,
+                       titleIconName: 'fa-globe',
+                       options: this.countries,
+                       helpMessage: gettext("The country that team members primarily identify with.")
+                   });
+               },
+
+               render: function() {
+                   this.$el.html(_.template(edit_team_template)({}));
+                   this.set(this.teamNameField, '.team-required-fields');
+                   this.set(this.teamDescriptionField, '.team-required-fields');
+                   this.set(this.optionalDescriptionField, '.team-optional-fields');
+                   this.set(this.teamLanguageField, '.team-optional-fields');
+                   this.set(this.teamCountryField, '.team-optional-fields');
+                   return this;
+               },
+
+               set: function(view, selector) {
+                   this.$el.find(selector).append(view.$el);
```

Comment:
HTML generated when `setElement` is used
```html
<div class="team-name-field">
	<label class="u-field-title" for="u-field-input-name">
    	Team Name (Required) *
	</label>
	<span class="u-field-value">
	    <input id="u-field-input-name" aria-describedby="u-field-message-help-name" type="text" name="input" value="">
	</span>
	<span class="u-field-message" id="u-field-message-name">
	    <span class="u-field-message-notification" aria-live="polite"></span>
	    <span class="u-field-message-help" id="u-field-message-help-name"> A name that identifies your team (maximum 255 characters).</span>
	</span>
</div>
```

HTML generated when `append` is used
```html
<div class="u-field u-field-text u-field-name editable-always mode-edit">
	<label class="u-field-title" for="u-field-input-name">
    	Team Name (Required) *
	</label>
	<span class="u-field-value">
	    <input id="u-field-input-name" aria-describedby="u-field-message-help-name" type="text" name="input" value="">
	</span>
	<span class="u-field-message" id="u-field-message-name">
	    <span class="u-field-message-notification" aria-live="polite"></span>
	    <span class="u-field-message-help" id="u-field-message-help-name"> A name that identifies your team (maximum 255 characters).</span>
	</span>
</div>
````

We are using fields from `fields.js`  they have their own html, in the case of `setElement` parent div is removed and i am not sure how we can use `highlightFieldOnError` and `unhighlightField` if we use `seElement`